pkg/provisioner/crdprovisioner.go

```diff
@@ -239,36 +228,55 @@ func (c *CrdProvisioner) CreateVolume(
 			},
 		}
 
-		azVolumeInstance, err := azVolumeClient.Create(ctx, azVolume, metav1.CreateOptions{})
+		workflow.Logger().Info("Creating AzVolume CRI")
+
+		_, err = azVolumeClient.Create(ctx, azVolume, metav1.CreateOptions{})
 		if err != nil {
-			klog.Errorf("Failed to create azvolume resource for volume name (%s), error: %v", volumeName, err)
+			workflow.Logger().Error(err, "failed to create AzVolume CRI")
 			return nil, err
 		}
 
-		if azVolumeInstance == nil {
-			return nil, status.Errorf(codes.Internal, "Failed to create azvolume resource volume name (%s)", volumeName)
-		}
+		workflow.Logger().Info("Successfully created AzVolume CRI")
+	}
 
-		klog.Infof("Successfully created AzVolume CRI (%s)...", azVolumeName)
+	var waiter *conditionWaiter
+	waiter, err = c.conditionWatcher.newConditionWaiter(ctx, azVolumeType, azVolumeName, func(obj interface{}, _ bool) (bool, error) {
+		if obj == nil {
+			return false, nil
+		}
+		azVolumeInstance := obj.(*diskv1beta1.AzVolume)
+		if azVolumeInstance.Status.Detail != nil {
+			return true, nil
+		} else if azVolumeInstance.Status.Error != nil {
+			return false, util.ErrorFromAzError(azVolumeInstance.Status.Error)
+		}
+		return false, nil
+	})
+	if err != nil {
+		return nil, err
 	}
+	defer waiter.Close()
 
-	obj, err := waiter.Wait(ctx)
+	var obj runtime.Object
+	obj, err = waiter.Wait(ctx)
 	if obj == nil || err != nil {
+		workflow.Logger().Error(err, "failed to create volume")
```

Comment:
This seems unnecessary. We should be able to infer the volume creation failure from the `Finish` log.